walletunlocker/service.go

```diff
@@ -434,11 +508,86 @@ func (u *UnlockerService) ChangePassword(ctx context.Context,
 			"%v", err)
 	}
 
+	// The next step is to load the macaroon database, change the password
+	// then close it again.
+	// Attempt to open the macaroon DB, unlock it and then change
+	// the passphrase.
+	macaroonService, err := macaroons.NewService(
+		netDir, "lnd", in.StatelessInit,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	err = macaroonService.CreateUnlock(&privatePw)
+	if err != nil {
+		closeErr := macaroonService.Close()
+		if closeErr != nil {
+			return nil, fmt.Errorf("could not create unlock: %v "+
+				"--> follow-up error when closing: %v", err,
+				closeErr)
+		}
+		return nil, err
+	}
+	err = macaroonService.ChangePassword(privatePw, in.NewPassword)
```

Comment:
Why do we need to now _explicitly_ perform this new password change at the macaroon service level? If we crash at this point, but the above succeeds, then the user may end up in a situation where they can't start up (?) or just aren't able to use their macaroons. I guess in either case, then they can just remove their on-disk macaroons, then have things just "work".

Reply:
Before we just deleted the whole macaroon DB. But since we now want to support only changing the password, without invalidating all macaroons, we have to perform the actual password change on the DB level.

Indeed, if something goes wrong here, it could result in the wallet DB password being changed but the macaroon DB password not. The user then still has the option to completely remove the macaroon DB and all macaroon files to resolve this. Not sure if we need to address this in this PR?